simple_clock/app.js

```diff
@@ -1,3 +1,106 @@
 window.addEventListener("load", function() {
-  console.log("Hello World!");
+  var intervalID = window.setInterval(updateTime, 1000);
+  initStopWatch();
+  initAlarm();
 });
+
+function initAlarm() {
+  var startAlarmBtn = document.getElementById('start-alarm');
+  startAlarmBtn.onclick = function () {
+    var delay = document.getElementById('delay');
+    var date = new Date();
+    date.setSeconds(date.getSeconds() + +delay.value);
+    var alarm = navigator.mozAlarms.add(date, 'ignoreTimezone');
+  }
+}
+
+navigator.mozSetMessageHandler("alarm", function (alarm) {
+  new Notification("Alarm!!!!");
+});
+
+function updateTime() {
+  var content = document.querySelector('content');
+  content.innerHTML = getNowTimeString();
+}
+
+function getNowTimeString() {
+  var now = new Date();
+  var year = now.getFullYear();
+  var month = now.getMonth();
+  var date = now.getDate();
+  var hour = now.getHours();
+  var minute = now.getMinutes();
+  var second = now.getSeconds();
+  var week = now.getWeekNumber();
+  var timezone = now.getTimezoneOffset();
+  var timezoneString = 'UTC';
+  if (timezone > 0) {
+    timezoneString += '-';
+  } else {
+    timezoneString += '+';
+  }
+
+  timezoneString = timezoneString + Math.abs(timezone) / 60 + 'hour ';
+  if (Math.abs(timezone) % 60 != 0) {
+    timezoneString = timezoneString + Math.abs(timezone) % 60 + 'min';
+  }
+
+  return year + '/' + pad(month, 2) + '/' + pad(date, 2) + ' ' + pad(hour, 2) + ':' + pad(minute, 2) + ':' + pad(second, 2) + '<br> Week: ' + week + ', Timezone: ' + timezoneString;
+}
+
+function pad(number, length) {
+  var str = '' + number;
+  while (str.length < length) {
+    str = '0' + str;
+  }
+  return str;
+}
+
+Date.prototype.getWeekNumber = function(){
```

Comment:
There are some popular libraries that will modify builtins' `prototype`, but this is not a good idea in the most cases.